addons/project_timesheet_holidays/tests/test_timesheet_holidays.py

```diff
@@ -127,3 +127,40 @@ def test_validate_without_timesheet(self):
         })
         holiday.with_user(SUPERUSER_ID).action_validate()
         self.assertEqual(len(holiday.timesheet_ids), 0, 'Number of generated timesheets should be zero since the leave type does not generate timesheet')
+
+    def test_timesheet_time_off_including_public_holiday(self):
+        """ Public holiday should not generate an entry if the employee is already on time off
+
+            Test Case:
+            =========
+            1) John Doe is on 1 week time off
+            2) Wednesday in the week is a public holiday
+            3) There will be five timesheets of timeoff. There will be 4 time off and 1 public holidays
+        """
+
+        leave_start_datetime = datetime(2022, 1, 24, 7, 0, 0, 0)# this is monday
+        leave_end_datetime = datetime(2022, 1, 28, 18, 0, 0, 0)
+
+        # public_holiday
+        self.env['resource.calendar.leaves'].create({
+            'name': 'Test',
+            'calendar_id': self.employee_working_calendar.id,
+            'date_from': datetime(2022, 1, 26, 7, 0, 0, 0),  #This is a wednesday and india Independence
+            'date_to': datetime(2022, 1, 26, 18, 0, 0, 0),
+        })
+
+        holiday = self.Requests.with_user(self.user_employee).create({
+            'name': 'Leave 1',
+            'employee_id': self.empl_employee.id,
+            'holiday_status_id': self.hr_leave_type_with_ts.id,
+            'date_from': leave_start_datetime,
+            'date_to': leave_end_datetime,
+        })
+        holiday.with_user(SUPERUSER_ID).action_validate()
+
+        timeoff_timesheet = self.env['account.analytic.line'].search_count([
+            ('date', '>=', leave_start_datetime),
+            ('date', '<=', leave_end_datetime),
+            ('employee_id', '=', self.empl_employee.id)])
+
+        self.assertEqual(timeoff_timesheet, 5)
```

Comment:
```suggestion
        self.assertEqual(timeoff_timesheet, 5)
        self.assertEqual(holiday.timesheet_ids, 4, '4 timesheets should be generated for this time off.')
```

Reply:
Why don't you add a assert to check how many timesheets have been generated by the leave taken by the employee? 🤔

Reply:
@xavierbol Now assert has checked how many timesheet created of global holiday and employee time-off.Thanks for the suggestion!